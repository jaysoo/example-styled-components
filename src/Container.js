import styled from 'styled-components';
import Message from './Message'

const Container = styled.div`
  font-weight: ${props => props.bold ? 'bold' : 'normal'};
  display: flex;
  justify-content: center;
  align-items: center;
  font-family: Helvetica, sans-serif;

  ${Message} {
    margin: 3px;
  }
`;

export default Container;
