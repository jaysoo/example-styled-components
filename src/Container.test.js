import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import Container from './Container';

test('normal', () => {
  const component = renderer.create(
    <Container>
      Hello
    </Container>
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
});

test('bold', () => {
  const component = renderer.create(
    <Container bold>
      Hello
    </Container>
  )
const tree = component.toJSON()
expect(tree).toMatchSnapshot()
});
