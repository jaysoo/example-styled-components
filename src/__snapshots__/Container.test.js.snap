// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bold 1`] = `
.c0 {
  font-weight: bold;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: Helvetica, sans-serif;
}

.c0 .sc-bdVaJa {
  margin: 3px;
}

<div
  className="c0"
>
  Hello
</div>
`;

exports[`normal 1`] = `
.c0 {
  font-weight: normal;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: Helvetica, sans-serif;
}

.c0 .sc-bdVaJa {
  margin: 3px;
}

<div
  className="c0"
>
  Hello
</div>
`;
